java/memory/src/main/java/io/netty/buffer/PooledByteBufAllocatorL.java

```diff
@@ -57,6 +59,17 @@ public PooledByteBufAllocatorL(MetricRegistry registry) {
     empty = new UnsafeDirectLittleEndian(new DuplicatedByteBuf(Unpooled.EMPTY_BUFFER));
   }
 
+  public static void setUpMnemonicUnpooledByteBufAllocator(MnemonicUnpooledByteBufAllocator<?> mubballocator) {
```

Comment:
Instead of statically setting an allocator, it should be an optional constructor parameter of type ByteBufAllocator.

Reply:
Got it, let me try to make it optional, Thanks.

Reply:
found the field INNER_ALLOCATOR of AllocationManager is initialized with "new PooledByteBufAllocatorL()" in
https://github.com/apache/arrow/blob/master/java/memory/src/main/java/org/apache/arrow/memory/AllocationManager.java#L68
so should I add an optional constructor for AllocationManager as well?

The bufferWithoutReservation(..) of BaseAllocator, in turn, instantiates and use the AllocationManager, 
https://github.com/apache/arrow/blob/2706b7fc5224ff56441d46eec516b2072090a984/java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java#L309

so is the method the best place to inject Mnemonic's allocator as its parameter?
Thanks!